Build Rnote and Installer for Windows Arm64

OB0NE opened the pull request:
Following @Doublonmousse 's advice in #1209 ,I made some adjustments to the former request #1209 to add support for building Rnote on Windows ARM64 while not breaking the x86 build process. architecture. The changes include:

1. Added a new Meson option 'win-arm64-build-environment-path' in meson_options.txt to specify the path for the ARM64 build environment. I also changed the meson.build to detect architecture and chose build environment accordingly.

2. Modified cargo_build.py to detect the system architecture and set appropriate Rust flags:
   - Added platform detection for Windows ARM64.
   - Set '-C linker=clang' flag for ARM64 builds on Windows.

3. Updated rnote_inno.iss.in to include ARM64 architecture support for the Windows installer. 

And the issue about black borders is fixed. See https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/7692.

**I’ve only been able to test the build on a Microsoft Surface Pro 11, so I'm unsure how it will perform on other ARM devices.**

flxzt commented:
Maybe I am wrong, but I don't directly see the need for the `win-arm64-build-environment-path` meson variable. Won't `win-build-environment-path` already cover this? One would just need to configure the project to point to the clangarm64 environment when building on ARM.

OB0NE commented:
> Maybe I am wrong, but I don't directly see the need for the `win-arm64-build-environment-path` meson variable. Won't `win-build-environment-path` already cover this? One would just need to configure the project to point to the clangarm64 environment when building on ARM.

Do you mean git rid of the `win-arm64-build-environment-path` , and build like this?
```
meson setup --prefix=/usr -Dwin-build-environment-path=C:\msys64\clangarm64 _mesonbuild
```

flxzt commented:
> Do you mean git rid of the `win-arm64-build-environment-path` , and build like this?
> 
> ```
> meson setup --prefix=/usr -Dwin-build-environment-path=C:\msys64\clangarm64 _mesonbuild
> ```

Yes exactly!

OB0NE commented:
> > Do you mean git rid of the `win-arm64-build-environment-path` , and build like this?
> > ```
> > meson setup --prefix=/usr -Dwin-build-environment-path=C:\msys64\clangarm64 _mesonbuild
> > ```
> 
> Yes exactly!

I've deleted the 'win-arm64-build-environment-path' and changed the build instructions.

flxzt commented on misc/building/rnote-windows-arm64-build.md:
````diff
@@ -0,0 +1,96 @@
+```
+export PATH=$PATH:/c/msys64/clangarm64/bin
+```
+If you have installed Inno Setup, add the following line to the end of the `C:\msys64\home\ZhouZhiwu\.bashrc` file:
````
this path is specific to your system, consider replacing the user part with `<user>` or something similar

flxzt commented on misc/building/rnote-windows-arm64-build.md:
```diff
@@ -0,0 +1,96 @@
+
+### Configuration
+
+Add the following line to the end of the `C:\msys64\home\ZhouZhiwu\.bashrc` file to add the Rust binary directory to MSYS2's `PATH`:
```
this path is specific to your system as well

OB0NE replied:
Sorry, I didn't notice this. I have changed it.

OB0NE commented on build-aux/cargo_build.py:
```diff
@@ -22,7 +23,16 @@
+else:
+    rustflags = ''
+
+cargo_call = f"env {cargo_env} RUSTFLAGS='{rustflags}' {cargo_cmd} build {cargo_options}"
```
Thank you for the feedback about moving the logic to the meson build file. I want to be transparent - I've been using AI to help me adjust the code as I have limited experience with Rust and the app building process. This approach has proven to be quite inefficient, and despite multiple attempts, I haven't been able to successfully move the logic to the meson build file.

Rather than continue with AI-assisted code adjustments that might not meet the project's standards, I think I can be more helpful by offering to test any changes you implement on my ARM64 device (Surface Pro 11). This would ensure the changes work correctly on ARM64 Windows hardware, which might be valuable for the project.

Let me know if you'd like me to test any changes you make, and I'll be happy to provide detailed feedback about how it works on ARM64.
